app/controllers/events_controller.rb

```diff
@@ -11,17 +9,19 @@ def all
   end
 
   def create
-    event = Apple.new.fire({ message: params[:message] })
-    publish event
 
-    event = Orange.new.receive event
-    publish event
+    originating_data = { message: params[:message] }
+
+    steps = [{ name: 'Apple',  type: 'EventHandler' },
+             { name: 'Orange', type: 'EventHandler' },
+             { name: 'Banana', type: 'EventHandler' },
+             { name: 'Pear',   type: 'EventHandler' }]
 
-    event = Banana.new.receive event
-    publish event
+    steps.each_with_index { |x, i| x[:method] = (i == 0 ? :fire : :receive) }
 
-    event = Pear.new.receive event
-    publish event
+    event = steps.reduce(originating_data) do |k, i|
+      i[:type].constantize.new.send(i[:method], k)
+    end
```

Comment:
@calebcauthon Of course, we'll need security around this.  We do not want to let inputs drive the creation of new objects.  "Insecure Deserialization" is #8 in the OWASP Top 10.  But we're building a flow right now.